tests/h/views/api/flags_test.py

```diff
@@ -8,105 +8,114 @@
 
     def test_passes_annotation_target_uri_to_flag_notification_email(self,
+                                                                     annotation_context,
                                                                      pyramid_request,
-                                                                     group_service,
+                                                                     groupfinder_service,
                                                                      flag_notification_email,
                                                                      incontext_link):
-        context = mock.Mock()
-        pyramid_request.json_body = {'annotation': context.annotation.id}
+        pyramid_request.json_body = {'annotation': annotation_context.annotation.id}
         incontext_link.return_value = None
 
-        views.create(context, pyramid_request)
+        views.create(annotation_context, pyramid_request)
 
         flag_notification_email.assert_called_once_with(request=pyramid_request,
-                                                        email=group_service.find.return_value.creator.email,
-                                                        incontext_link=context.annotation.target_uri)
+                                                        email=groupfinder_service.find.return_value.creator.email,
+                                                        incontext_link=annotation_context.annotation.target_uri)
 
     def test_sends_notification_email(self,
+                                      annotation_context,
                                       pyramid_request,
                                       flag_notification_email,
                                       mailer):
-        context = mock.Mock()
-        pyramid_request.json_body = {'annotation': context.annotation.id}
+        pyramid_request.json_body = {'annotation': annotation_context.annotation.id}
+
+        views.create(annotation_context, pyramid_request)
 
-        views.create(context, pyramid_request)
         mailer.send.delay.assert_called_once_with(*flag_notification_email.return_value)
 
     def test_doesnt_send_email_if_group_has_no_creator(self,
+                                                       annotation_context,
                                                        factories,
-                                                       group_service,
+                                                       groupfinder_service,
                                                        pyramid_request,
                                                        mailer):
-        annotation_context = mock.create_autospec(AnnotationContext, instance=True, annotation=factories.Annotation())
-        group_service.find.return_value = factories.Group(creator=None, members=[])
+        groupfinder_service.find.return_value = factories.Group()
+        groupfinder_service.find.return_value.creator = None
```

Comment:
Could this line by moved into a `creator=None` kwarg to the `factories.Group` constructor?